src/main/java/uk/gov/hmcts/reform/sscs/service/exceptions/FeignClientErrorDecoder.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.sscs.service.exceptions;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Request;
+import feign.Request.HttpMethod;
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.hmcts.reform.sscs.model.HmcFailureMessage;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingRequestPayload;
+import uk.gov.hmcts.reform.sscs.service.AppInsightsService;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+
+@Slf4j
```

Comment:
This class needs some form of integration testing to verify behaviour with error responses from the Feign client